File: internal/gateway/link.go
package gateway

import (
	"github.com/chirino/graphql/schema"
	"github.com/pkg/errors"
)

type Link struct {
	Field       string            `yaml:"field,omitempty"`
	Description string            `yaml:"description,omitempty"`
	Upstream    string            `yaml:"upstream,omitempty"`
	Query       string            `yaml:"query,omitempty"`
	Vars        map[string]string `yaml:"vars,omitempty"`
}

func (c actionRunner) link(action *Link) error {
	endpoint, ok := c.Endpoints[action.Upstream]
	if !ok {
		return errors.New("invalid endpoint id: " + action.Upstream)
	}
	if action.Field == "" {
		return errors.New("field must be set")
	}
	field := schema.Field{Name: action.Field}
	if action.Description != "" {
		field.Desc = schema.Description{Text: action.Description}
	}
	return mount(c, field, endpoint, action.Query, action.Vars)
}

File: ui/src/pages/Schema/Link.js
import {FormGroup, SelectOption, TextArea, TextInput} from '@patternfly/react-core';

import React from 'react';
import BetterSelect from "../../components/BetterSelect";
import {fieldSetters} from "../../utils";
import VarList from "./VarList";

const Link = ({upstreams, action, setAction}) => {

  const onChange = fieldSetters(Object.assign({
    field: "",
    upstream: "",
    query: "",
    vars: []
  }, action), setAction)

  return <React.Fragment>
    <FormGroup label="Field" fieldId="field" isRequired
               helperText="The new field name that will be created on the gateway schema">
      <TextInput
        id="field" name="field"
        value={action.field} onChange={onChange.field}
        isRequired type="text"
      />
    </FormGroup>

    <FormGroup label="Upstream Server" isRequired fieldId="upstream" helperText="The upstream sever to send the request to">
      <BetterSelect value={action.upstream} setValue={onChange.upstream}>
        {upstreams.map((item, key) =>
          <SelectOption value={item.name} key={key}>{item.name}</SelectOption>
        )}
      </BetterSelect>
    </FormGroup>

    <FormGroup label="Query" isRequired fieldId="query" helperText="A partial upstream GraphQL query that requests will be routed to">
      <TextArea
        id="query" name="query"
        value={action.query} onChange={onChange.query}
        isRequired type="text"
      />
    </FormGroup>

    <VarList value={action.vars} onChange={onChange.vars}>test</VarList>

  </React.Fragment>
}

export default Link